drivers/usb/host/xhci-mem.c

```diff
@@ -2504,13 +2506,13 @@ int xhci_mem_init(struct xhci_hcd *xhci, gfp_t flags)
 		goto fail;
 
 	/* set ERST count with the number of entries in the segment table */
-	val = readl(&xhci->ir_set->erst_size);
-	val &= ERST_SIZE_MASK;
-	val |= ERST_NUM_SEGS;
+	val2 = readl(&xhci->ir_set->erst_size);
+	val2 &= ERST_SIZE_MASK;
+	val2 |= val;
 	xhci_dbg_trace(xhci, trace_xhci_dbg_init,
 			"// Write ERST size = %i to ir_set 0 (some bits preserved)",
 			val);
```

Comment:
The swapping of val and val2 (just to use them in order?) is a bit gratuitous, and it breaks/non-obviously fixes this debug output.

Reply:
Bits 31:16 are RsvdP (reserved, preserve anyway) but this now truncates the output if any of the reserved bits are used. Debug should use val2 - will fix

Reply:
And swap val and val2? Plz?

Reply:
k

Reply:
Thx